lib/spack/docs/tutorial_buildsystems.rst

```diff
@@ -0,0 +1,351 @@
+    ==> This looks like a URL for bowtie
+    ==> Found 1 version of bowtie:
+
+    1.2.1.1  https://downloads.sourceforge.net/project/bowtie-bio/bowtie/1.2.1.1/bowtie-1.2.1.1-src.zip
+
+    ==> How many would you like to checksum? (default is 1, q to abort) 1
+    ==> Downloading...
+    ==> Fetching https://downloads.sourceforge.net/project/bowtie-bio/bowtie/1.2.1.1/bowtie-1.2.1.1-src.zip
+    ######################################################################## 100.0%
+    ==> Checksummed 1 version of bowtie
+    ==> This package looks like it uses the makefile build system
+    ==> Created template for bowtie package
+    ==> Created package file: /Users/mamelara/spack/var/spack/repos/builtin/packages/bowtie/package.py
+
+Once the fetching is completed, Spack will open up your text editor in the
+usual fashion and create a template of a MakefilePackage package.py.
+
+.. literalinclude:: tutorial/examples/Makefile/0.package.py
+   :language: python
+
+Spack was successfully able to detect that bowtie uses GNU Make. Let's add
+in the rest of our details for our package:
+
+.. literalinclude:: tutorial/examples/Makefile/1.package.py
+   :language: python
+
+If the application is simple to build then the contents present in our package.py
+is enough and we don't have to override any methods in our Makefile package. In
+fact, for most of the build systems, overriding methods is optional and packages
+can be kept as simple as what we have already. 
+
+Unfortunately, the defaults that bowtie sets up are not what we 
+need. If we look inside the Makefile, we see that CC and CXX point to our GNU
+compiler:
+
+.. code-block:: make
+    
+    CPP = g++ -w
+    CXX = $(CPP)
+    CC = gcc
+    LIBS = $(LDFLAGS) -lz
+    HEADERS = $(wildcard *.h)
+
+To fix this we need to use the edit method which is used to write to our custom
+Makefile:
+
+.. literalinclude:: tutorial/examples/Makefile/2.package.py
+   :language: python
+
+Here we used regex to make CC and CXX point at whatever Spack sets
+CC and CXX to. This allows us to build bowtie with whatever compiler we specify
+through Spack's specs. We can also specify our build targets and install targets
+usng the build() and install() methods.
+
+.. literalinclude:: tutorial/examples/Makefile/3.package.py
+   :language: python
+
+In our build stage we are able to provide an argument to make whether we
+decide to build with tbb support or not. For the install, phase we provide
+the prefix and then invoke make install. 
+
+Through this example, we are able to see that we can change the Makefile through
+our edit method, as well as invoke build, and install to specify our build
+and install targets, respectively.
+
+-------------------
+Autotools
+-------------------
+
+As mentioned earlier, Autotools consist of three parts:  autoconf, automake,
+and libtool. Your typical build incantation will consist of the following:
+
+.. code-block:: console
+
+    ./configure --prefix=/path/to/install_dir
```

Comment:
Do we want to show this in bash (like here) or in python as it would be in the install method? Then they can see how it compares to the package we used in the first part of the tutorial.